src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs

```diff
@@ -25,8 +27,36 @@ public class InstrumentFieldsMiddleware : IFieldMiddleware
                 { "path", context.ResponsePath },
             };
 
-            using (context.Metrics.Subject("field", name, metadata))
-                return await next(context).ConfigureAwait(false);
+            var marker = context.Metrics.Subject("field", name, metadata);
+            var ret = await next(context).ConfigureAwait(false);
```

Comment:
Can this throw?

Reply:
Yes. Looks like it needs a try catch block…